lib/rucio/core/rule.py

```diff
@@ -1157,13 +1365,34 @@ def repair_rule(rule_id: str, *, session: "Session", logger: LoggerFunction = lo
             insert_rule_history(rule=rule, recent=True, longterm=False, session=session)
             # Try to update the DatasetLocks
             if rule.grouping != RuleGrouping.NONE:
-                session.query(models.DatasetLock).filter_by(rule_id=rule.id).update({'state': LockState.STUCK})
+                stmt = update(
+                    models.DatasetLock
+                ).where(
+                    models.DatasetLock.rule_id == rule.id
+                ).values({
+                    models.DatasetLock.state: LockState.STUCK
+                })
+                session.execute(stmt)
             logger(logging.DEBUG, '%s while repairing rule %s', type(error).__name__, rule_id)
             return
 
         # Delete Datasetlocks which are not relevant anymore
-        validated_datasetlock_rse_ids = [rse_id[0] for rse_id in session.query(models.ReplicaLock.rse_id).filter(models.ReplicaLock.rule_id == rule.id).group_by(models.ReplicaLock.rse_id).all()]
-        dataset_locks = session.query(models.DatasetLock).filter_by(rule_id=rule.id).all()
+        stmt = select(
+            models.ReplicaLock.rse_id
+        ).where(
+            models.ReplicaLock.rule_id == rule.id
+        ).group_by(
+            models.ReplicaLock.rse_id
+        )
+        results = session.execute(stmt).all()
+        validated_datasetlock_rse_ids = [rse_id.rse_id for rse_id in results]
```

Comment:
1. No real reason other than personal coding style, will follow this convention from now on.
2. Agree, changed to use distinct instead:

`stmt = select(
            models.ReplicaLock.rse_id
        ).distinct(
        ).where(
            models.ReplicaLock.rule_id == rule.id
        )`

Reply:
My reasoning is to avoid having one-time-use intermediate variables unless there is a particular reason (e.g. it helps break down a long chain of operations).